client/document/index.jsx

```diff
@@ -23,19 +23,35 @@ import WordPressLogo from 'components/wordpress-logo';
 import { jsonStringifyForHtml } from '../../server/sanitize';
 
 class Document extends React.Component {
+	embedLocaleScript() {
+		const { lang, langRevisions = {}, i18nLocaleScript } = this.props;
+
+		if ( ! i18nLocaleScript ) {
+			return null;
+		}
+
+		const langRevision = langRevisions[ lang ];
+
+		return langRevision ? (
+			<script src={ i18nLocaleScript + `?v=${ langRevision }` } />
+		) : (
+			<script src={ i18nLocaleScript } />
+		);
+	}
+
```

Comment:
I'd rather avoid having a pretty special method like this in `client/document/index` (among other things, we'll probably want to add this script to other templates, such as `client/document/desktop`). All we really need here is the URL for the `<script />` `src`. Let's compute that URL in `server/pages` or `server/render` (probably the latter, since we're setting `i18nLocaleScript` there now).